### ISSUE TITLE
After saving a .js file, the dialog "This file has been modified by another program.." opens

### ISSUE THREAD

**Issue by beholder4096** [notepad-plus-plus/notepad-plus-plus#5941]
### Description of the Issue

Guys, please help me with this, I am at the of my rope here.

When I open an existing javascript file in a new single instance of N++, change it, save it, switch windows app to another and then switch back, a dialog appears that the JS file has been changed by another program (although it was NOT, I checked via Total Commander file compare). 

This started happening after upgrading to newest N++, the old 6.9.9 does not do this, I checked it from backup.

### Steps to Reproduce the Issue

1. Have a js file ready, or just rename any file to .js extension
2. open the file in latest N++
3. change the file and save it
4. switch via Alt+Tab to another app on the desktop
5. switch back to N++

### Expected Behavior

The file should display normally, no dialogs or anything.

### Actual Behavior

The prompt "File has been changed by another program.." appears. The dirty mark of the file is reenabled when I press "no" to the reload prompt.

### Debug Information

Notepad++ v7.7.1   (32-bit)
Build time : Jun 16 2019 - 21:14:50
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : ON
Local Conf mode : OFF
OS : Windows 7 (32-bit)
Plugins : mimeTools.dll NppConverter.dll NppExport.dll 

### Additional Information

- The file must always be newly loaded into a single instance of N++ to reproduce the bug, always close N++ before retesting.
- only *.js files are affected, I tested also .txt and .php and nothing.
- I also tested the standalone portable newest version, also affected.
- Similar bug may also happen with js files under different circumstances, right after I save a file, literally in the same second, didn't try to reproduce that one.
- This behavior does not happen under 6.9.9, tested.  I have to switch to the older version because I work with JS a lot.
- **Please don't tell me to enable silent updates of the files.**

**Comment by beholder4096** [comment]
### More info ### 
Could not leave it like that because of that association with file extension (js). So I tried to change the file language in N++ when a test JS file was loaded. Changed it to anything arbitrary, changed it back to JavaScript. AND IT STOPPED DOING THE FILE CHANGED PROMPT for that testfile.js file. When I changed the filename of the test file (not the .js extension), the bug resurfaced.

So this is something that N++ saves with the file. Not inside the file but perhaps to N++ settings somewhere (about that particular file). Like which file belongs to which language or something. If it finds the language info about the file, the bug never shows. If not, it shows. 

Still have no idea how come only the js extension be associated with this weird bug.

**Comment by beholder4096** [comment]
It's safer to **always test this issue with a completely new filename** (with a .js) extension, just to ensure reproducibility of the bug. I have found out that the stand-alone installation of N++ "remembers" something about a file it had once opened (although it did show the bug once, it may not be reproducible for the same filename again).

**Comment by sasumner** [comment]
Did some experimentation with this and could not reproduce.

Notepad++ v7.7.1   (32-bit)
Build time : Jun 16 2019 - 21:14:50
Path : C:\.........\npp.7.7.1.bin\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 10 (64-bit)
Plugins : ColumnTools.dll Explorer.dll LocationNavigate.dll MarkdownViewerPlusPlus.dll mimeTools.dll NppConverter.dll NppExec.dll NppExport.dll PythonScript.dll

**Comment by beholder4096** [comment]
Could not be reproduced on WinXP Home SP3 NTFS partition.
Could not be reproduced on Win7 Enterprise SP1 FAT32 partition
**Only** reproducible on a Win7 Enterprise SP1 system, on any NTFS partition.

**Comment by beholder4096** [comment]
So I could not leave it like that anymore so I ran Process Monitor with filter on a test folder to see what is happening to a single file in that folder. It turned out that cmdagent.exe (security software Comodo Firewall suite) was accessing/resaving each new *.js file on an NTFS partition which was being saved by Notepad++. 

I first turned off cmdagent.exe process from working memory just to be sure it is the root of the issue. Indeed it was. Then I ran it again and disabled several setting which seemed to remove the issue altogether. I cannot seem to be able to reenable the issue (it's not accessing JS files anymore). I'll keep watching it just to be sure, maybe it needs a restart to begin doing it again.
